src/transformers/models/blip/modeling_blip.py

```diff
@@ -0,0 +1,1453 @@
+            encoder_hidden_states=image_embeds,
+            labels=decoder_targets,
+            return_dict=return_dict,
+        )
+
+        if not return_dict:
+            outputs = (outputs[0], outputs[1], image_embeds, vision_outputs[0]) + vision_outputs[2:]
+            return tuple(output for output in outputs if output is not None)
+
+        return BlipForConditionalGenerationModelOutput(
+            loss=outputs.loss,
+            decoder_logits=outputs.logits,
+            image_embeds=image_embeds,
+            last_hidden_state=vision_outputs.last_hidden_state,
+            hidden_states=vision_outputs.hidden_states,
+            attentions=vision_outputs.attentions,
+        )
+
+    @torch.no_grad()
+    def generate(
+        self,
+        pixel_values: torch.FloatTensor,
+        input_ids: Optional[torch.LongTensor] = None,
+        pixel_mask: Optional[torch.LongTensor] = None,
+        token_type_ids: Optional[torch.LongTensor] = None,
+        attention_mask: Optional[torch.LongTensor] = None,
+        **generate_kwargs
+    ) -> torch.LongTensor:
+        r"""
+        Overrides *generate* function to be able to use the model as a conditional generator
+
+        Parameters:
+            pixel_values (*torch.FloatTensor* of shape *(batch_size, image_width, image_height)*:
+                Input image to be processed
+            input_ids (*torch.LongTensor* of shape *(batch_size, sequence_length)*, *optional*):
+                The sequence used as a prompt for the generation.
+            pixel_mask (*torch.LongTensor* of shape *(batch_size, image_width, image_height)*, *optional*):
+                Mask to be used for the image - not used but kept for compatibility with the *BlipProcessor*
+            token_type_ids (*torch.LongTensor* of shape *(batch_size, sequence_length)*, *optional*):
+                Segment token indices to indicate first and second portions of the inputs. Not used but kept for
+                compatibility with the *BlipProcessor*
+            attention_mask (*torch.LongTensor* of shape *(batch_size, sequence_length)*, *optional*):
+                Mask to avoid performing attention on padding token indices. Mask values selected in `[0, 1]`:
+
+
+        Examples:
+        ```python
+        >>> from PIL import Image
+        >>> import requests
+        >>> from transformers import BlipProcessor, BlipForImageCaptioning
+
+        >>> model = BlipForImageCaptioning.from_pretrained("Salesforce/blip-image-captioning-base")
+        >>> processor = BlipProcessor.from_pretrained("Salesforce/blip-image-captioning-base")
+
+        >>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+        >>> image = Image.open(requests.get(url, stream=True).raw)
+
+        >>> inputs = processor(images=image, return_tensors="pt")
+
+        >>> outputs = model(**inputs)
+        >>> image_embeds = outputs.image_embeds
+        ```
+        """
+
+        batch_size, _ = pixel_values.shape[:2]
```

Comment:
```suggestion
        batch_size = pixel_values.shape[0]
```
I would also move this below the `if input_ids is None` block, as it's only then that it's needed